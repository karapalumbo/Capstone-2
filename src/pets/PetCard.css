.favorite-btn {
  float: right;
}

img {
  width: 346px;
  height: 346px;
  border-radius: 20px 20px 0px 0px;
}

.petcard {
  margin: 20px;
  border-radius: 20px 20px 0px 0px !important;
}

.petcard-body {
  padding: 0px !important;
}

.petcard-info {
  text-align: left;
  padding: 20px;
}
