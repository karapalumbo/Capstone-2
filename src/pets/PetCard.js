import { useContext, useState, useEffect } from "react";
import UserContext from "../UserContext";
import { Card, Button, CardBody } from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faHeart } from "@fortawesome/free-regular-svg-icons";
import { faHeart as solidHeart } from "@fortawesome/free-solid-svg-icons";
import PetfinderApi from "../api/api";
import Loading from "../Loading";
import "./PetCard.css";

import PetModal from "./PetModal";

const PetCard = ({
  pet_id,
  name,
  species,
  age,
  gender,
  description,
  color,
  photos,
  organizationID,
}) => {
  const { hasFavorited, favoritePet, unfavoritePet } = useContext(UserContext);

  const [favoritedPet, setFavoritedPet] = useState(false);
  const [isModalOpen, setModalOpen] = useState(false);
  const [org, setOrg] = useState(null);

  useEffect(
    function updateFavorited() {
      setFavoritedPet(hasFavorited(pet_id));
    },
    [pet_id, hasFavorited]
  );

  const getPetOrgInfo = async () => {
    const organization = await PetfinderApi.getOrganization(organizationID);
    setOrg(organization);
  };

  useEffect(function getPetOrg() {
    getPetOrgInfo();
  }, []);

  const handleFavorited = async () => {
    if (favoritedPet) {
      unfavoritePet(pet_id);
      setFavoritedPet(false);
    } else {
      setFavoritedPet(true);
      favoritePet(pet_id);
    }
  };

  const handleModal = () => {
    setModalOpen(!isModalOpen);
  };

  if (!org) return <Loading />;

  return (
    <Card className="petcard">
      <CardBody className="petcard-body">
        <img src={photos} alt={species} />

        <div className="petcard-info">
          <h4>{name}</h4>

          <span className="pet-favorite">
            <Button
              color="white"
              className="favorite-btn"
              onClick={handleFavorited}
            >
              <FontAwesomeIcon
                id={favoritedPet ? "favorites" : "not-favorites"}
                size="2x"
                icon={favoritedPet ? solidHeart : faHeart}
              />
            </Button>
          </span>

          <div className="pet-details">
            {color} {species}
          </div>

          <div className="pet-details">{age}</div>

          <div className="pet-details">{gender}</div>

          <Button block className="pet-modal-btn" onClick={handleModal}>
            Adopt me!
          </Button>

          <PetModal
            modalOpen={isModalOpen}
            toggleModal={handleModal}
            pet_id={pet_id}
            name={name}
            description={description}
            org={org}
          />
        </div>
      </CardBody>
    </Card>
  );
};

export default PetCard;
