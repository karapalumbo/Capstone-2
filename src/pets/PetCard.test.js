import React from "react";
import { render } from "@testing-library/react";
import PetCard from "./PetCard";
import { MemoryRouter } from "react-router";
import UserTestProvider from "../testHelper";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserTestProvider>
        <PetCard
          key="123"
          pet_id="123"
          name="testpet"
          species="dog"
          age="young"
          gender="male"
          color="black"
          description="test description"
          photos=""
        />
      </UserTestProvider>
    </MemoryRouter>
  );
});

it("renders a pet being favorited", function () {
  const component = render(
    <MemoryRouter>
      <UserTestProvider hasFavorited={() => true}>
        <PetCard
          key="123"
          pet_id="123"
          name="testpet"
          species="dog"
          age="young"
          gender="male"
          color="black"
          description="test description"
          photos=""
        />
      </UserTestProvider>
    </MemoryRouter>
  );
  expect(component).toMatchSnapshot();
});

it("renders a pet being unfavorited", function () {
  const component = render(
    <MemoryRouter>
      <UserTestProvider unfavoritePet={() => true}>
        <PetCard
          key="123"
          pet_id="123"
          name="testpet"
          species="dog"
          age="young"
          gender="male"
          color="black"
          description="test description"
          photos=""
        />
      </UserTestProvider>
    </MemoryRouter>
  );
  expect(component).toMatchSnapshot();
});
