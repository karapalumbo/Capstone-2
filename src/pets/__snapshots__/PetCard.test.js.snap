// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a pet being favorited 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="petcard card"
      >
        <div
          class="petcard-body card-body"
        >
          <img
            alt="dog"
            src=""
          />
          <div
            class="petcard-info"
          >
            <h4>
              testpet
            </h4>
            <span
              class="pet-favorite"
            >
              <button
                class="favorite-btn btn btn-white"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-heart fa-w-16 fa-2x "
                  data-icon="heart"
                  data-prefix="fas"
                  focusable="false"
                  id="favorites"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </span>
            <div
              class="pet-details"
            >
              black
               
              dog
            </div>
            <div
              class="pet-details"
            >
              young
            </div>
            <button
              class="pet-modal-btn btn btn-secondary btn-block"
              type="button"
            >
              Learn More!
            </button>
            <div />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="petcard card"
    >
      <div
        class="petcard-body card-body"
      >
        <img
          alt="dog"
          src=""
        />
        <div
          class="petcard-info"
        >
          <h4>
            testpet
          </h4>
          <span
            class="pet-favorite"
          >
            <button
              class="favorite-btn btn btn-white"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-heart fa-w-16 fa-2x "
                data-icon="heart"
                data-prefix="fas"
                focusable="false"
                id="favorites"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </span>
          <div
            class="pet-details"
          >
            black
             
            dog
          </div>
          <div
            class="pet-details"
          >
            young
          </div>
          <button
            class="pet-modal-btn btn btn-secondary btn-block"
            type="button"
          >
            Learn More!
          </button>
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders a pet being unfavorited 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="petcard card"
      >
        <div
          class="petcard-body card-body"
        >
          <img
            alt="dog"
            src=""
          />
          <div
            class="petcard-info"
          >
            <h4>
              testpet
            </h4>
            <span
              class="pet-favorite"
            >
              <button
                class="favorite-btn btn btn-white"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-heart fa-w-16 fa-2x "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  id="not-favorites"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </span>
            <div
              class="pet-details"
            >
              black
               
              dog
            </div>
            <div
              class="pet-details"
            >
              young
            </div>
            <button
              class="pet-modal-btn btn btn-secondary btn-block"
              type="button"
            >
              Learn More!
            </button>
            <div />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="petcard card"
    >
      <div
        class="petcard-body card-body"
      >
        <img
          alt="dog"
          src=""
        />
        <div
          class="petcard-info"
        >
          <h4>
            testpet
          </h4>
          <span
            class="pet-favorite"
          >
            <button
              class="favorite-btn btn btn-white"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-heart fa-w-16 fa-2x "
                data-icon="heart"
                data-prefix="far"
                focusable="false"
                id="not-favorites"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </span>
          <div
            class="pet-details"
          >
            black
             
            dog
          </div>
          <div
            class="pet-details"
          >
            young
          </div>
          <button
            class="pet-modal-btn btn btn-secondary btn-block"
            type="button"
          >
            Learn More!
          </button>
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
